File: ros_noetic/cpp/src/subscriber/subscriber_int.cpp
#include "ros/ros.h"
#include "std_msgs/Int64.h"

void chatterIntCallback(const std_msgs::Int64::ConstPtr& msg)
{
  ROS_INFO("The number : [%i]", msg->data);
}

int main(int argc, char **argv)
{
 
  ros::init(argc, argv, "subscriber_int");

  ros::NodeHandle n;

  ros::Subscriber sub = n.subscribe("chatter/int", 1000, chatterIntCallback);

  ros::spin();

  return 0;
}

File: ros_noetic/python/src/subscriber/subscriber_int.py
#!/usr/bin/env python3

import rospy
from std_msgs.msg import Int32

def callback(data):
    rospy.loginfo(rospy.get_caller_id() + "The number = %i ", data.data)
    
def subscriber_int():
    rospy.init_node('subscriber_int', anonymous=True)
    rospy.Subscriber("chatter/int", Int32, callback)
    rospy.spin()

if __name__ == '__main__':
    subscriber_int()
